runtime/server/queries_metrics.go

```diff
@@ -456,3 +476,36 @@ func lookupMetricsView(ctx context.Context, rt *runtime.Runtime, instanceID, nam
 
 	return res, mv.State, nil
 }
+
+type decodedTimeRangeSummary struct {
+	Min, Max, Watermark string
+}
+
+func decodeTimeRangeSummary(row map[string]any) (*runtimev1.TimeRangeSummary, error) {
+	timeRangeSummary := decodedTimeRangeSummary{} // TODO: move this to a good place
+	err := mapstructure.Decode(row, &timeRangeSummary)
```

Comment:
Have you tried using `WeakDecode`? Then I think it would be able to directly decode ISO strings into `time.Time` types (but I may be wrong)

Reply:
It wont work by itself. We need to create a decoder with `DecoderConfig` and `StringToTimeHookFunc`. I just reused the code from `metricsview_aggregation` since we will also need for rilltime resolver.